dhis-2/dhis-services/dhis-service-node/src/main/java/org/hisp/dhis/fieldfilter/DefaultFieldFilterService.java

```diff
@@ -277,7 +285,36 @@ private AbstractNode buildNode( FieldMap fieldMap, Class<?> klass, Object object
             Object returnValue = ReflectionUtils.invokeMethod( object, property.getGetterMethod() );
             Class<?> propertyClass = property.getKlass();
             Schema propertySchema = schemaService.getDynamicSchema( propertyClass );
-            if ( returnValue != null && propertySchema.getProperties().isEmpty() && !property.isCollection() && property.getKlass().isInterface() && !property.isIdentifiableObject() )
+
+            if ( property.hasPropertyTransformer() )
+            {
+                PropertyTransformer propertyTransformer;
+
+                try
+                {
+                    propertyTransformer = property.getPropertyTransformer().newInstance();
```

Comment:
Consider using a "caching" layer here to avoid using reflection, which in this specific case could happen quite often, since the User is often part of payloads.